### PR TITLE
Update solvers interface

### PR DESCRIPTION
**Pull request by mforets**
Prep for https://github.com/ONSAS/ONSAS.jl/issues/216

The failure in `solve` might be analyzed separately.

### CODE REVIEW

**Review by mvanzulli — APPROVED** [review]

File: src/StructuralSolvers/StructuralSolvers.jl [hunk 1]
```diff
@@ -177,19 +175,33 @@ include("./NewtonRaphson.jl")
...
+"""
+Solve a structural analysis problem with the given solver.
+This function mutates the state defined in the `analysis` problem; use [`solve`](@ref) to avoid mutation.
+For linear analysis problems, `alg` doesn't need to be provided.
+
+Returns a solution structure holding the result and the algorithm used to obtain it.
+"""
+function solve!(analysis::A, alg::Union{AbstractSolver,Nothing}=nothing, args...;
+                kwargs...) where {A}
+    # TODO Dispatch on `nothing` only for linear problems.
```
Comment (mvanzulli) [inline src/StructuralSolvers/StructuralSolvers.jl:198]: Good idea 👍🏼

File: src/StructuralAnalyses/NonLinearStaticAnalyses.jl [hunk 2]
```diff
@@ -49,7 +49,7 @@ function Base.show(io::IO, sa::NonLinearStaticAnalysis)
     show(io, sa.state)
 end
 
-"Solves an `NonLinearStaticAnalysis` `sa` with an AbstractSolver `alg`."
+"Solves an `NonLinearStaticAnalysis` problem with a given solver."
```
Comment (mvanzulli) [inline src/StructuralAnalyses/NonLinearStaticAnalyses.jl:52]: I have a question, in general we want to use `  only for types or variable names too ?

**Review by mvanzulli — APPROVED** [review]

File: src/StructuralAnalyses/NonLinearStaticAnalyses.jl (hunk 2 above)
Comment (mforets) [reply src/StructuralAnalyses/NonLinearStaticAnalyses.jl:52]: usually i use it for code in general, so both

Comment (mvanzulli) [reply src/StructuralAnalyses/NonLinearStaticAnalyses.jl:52]: But i mean in the function docstrings, like this function apply the `BoundaryCondition` `bc` to the `Mesh` `m`. 
```julia
apply(m::Mesh, bc::BoundaryCondtion)
```

**Pull request merged**